internal/resource/config/password_credential_validators_resource.go

```diff
@@ -0,0 +1,649 @@
+	pluginDescRefId := plan.PluginDescriptorRef.Attributes()["id"].(types.String).ValueString()
+	pluginDescRefResLink := client.NewResourceLinkWithDefaults()
+	pluginDescRefResLink.Id = pluginDescRefId
+	pluginDescRefErr := json.Unmarshal([]byte(internaljson.FromValue(plan.PluginDescriptorRef, false)), pluginDescRefResLink)
+	if pluginDescRefErr != nil {
+		resp.Diagnostics.AddError("Failed to build plugin descriptor ref request object:", pluginDescRefErr.Error())
+		return
+	}
+
+	// Configuration
+	configuration := client.NewPluginConfiguration()
+	configErr := json.Unmarshal([]byte(internaljson.FromValue(plan.Configuration, true)), configuration)
+	if configErr != nil {
+		resp.Diagnostics.AddError("Failed to build plugin configuration request object:", configErr.Error())
+		return
+	}
+
+	updatePasswordCredentialValidators := r.apiClient.PasswordCredentialValidatorsApi.UpdatePasswordCredentialValidator(ProviderBasicAuthContext(ctx, r.providerConfig), plan.Id.ValueString())
+	createUpdateRequest := client.NewPasswordCredentialValidator(plan.Id.ValueString(), plan.Name.ValueString(), *pluginDescRefResLink, *configuration)
+	err := addOptionalPasswordCredentialValidatorsFields(ctx, createUpdateRequest, plan)
+	if err != nil {
+		resp.Diagnostics.AddError("Failed to add optional properties to add request for PasswordCredentialValidators", err.Error())
+		return
+	}
+	requestJson, err := createUpdateRequest.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Update request: "+string(requestJson))
+	}
+	updatePasswordCredentialValidators = updatePasswordCredentialValidators.Body(*createUpdateRequest)
+	updatePasswordCredentialValidatorsResponse, httpResp, err := r.apiClient.PasswordCredentialValidatorsApi.UpdatePasswordCredentialValidatorExecute(updatePasswordCredentialValidators)
+	if err != nil {
+		ReportHttpError(ctx, &resp.Diagnostics, "An error occurred while updating PasswordCredentialValidators", err, httpResp)
+		return
+	}
+	// Log response JSON
+	responseJson, err := updatePasswordCredentialValidatorsResponse.MarshalJSON()
+	if err == nil {
+		tflog.Debug(ctx, "Read response: "+string(responseJson))
+	}
+	// Read the response
+	readPasswordCredentialValidatorsResponse(ctx, updatePasswordCredentialValidatorsResponse, &plan, plan.Configuration)
+
+	// Update computed values
+	diags = resp.State.Set(ctx, plan)
+	resp.Diagnostics.Append(diags...)
+}
+
+// // Delete deletes the resource and removes the Terraform state on success.
+func (r *passwordCredentialValidatorsResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var state passwordCredentialValidatorsResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+	httpResp, err := r.apiClient.PasswordCredentialValidatorsApi.DeletePasswordCredentialValidator(ProviderBasicAuthContext(ctx, r.providerConfig), state.Id.ValueString()).Execute()
+	if err != nil {
```

Comment:
This should ignore errors that signify the resource doesn't exist: https://developer.hashicorp.com/terraform/plugin/framework/resources/delete#recommendations
Also the return isn't needed here